commands/logs.go

```diff
@@ -15,46 +16,74 @@ const maxNumberLogs = 1500
-	Length          int    `default:100 help:"Number of log lines to fetch."`
+	AccountID    int    `required short:"a" help:"ID of account to query"`
+	AppID        int    `required short:"i" help:"ID of app to query"`
+	AppPath      string `default:"nodejs" help:"Path of NodeJS application in environment repository."`
+	InstanceName string `default:"" help:"Specific instance of NodeJS application running on Section platform."`
+	Length       int    `default:100 help:"Number of log lines to fetch."`
+	Follow       bool   `help:"Displays recent logs and leaves the session open for logs to stream in. --instance-name required."`
 	// StartTimestamp  int    `default:0 help:"Start of log time stamp to fetch."`
 	// EndTimestamp    int    `default:0 help:"End of log time stamp to fetch."`
 }
 
 // Run executes the command
 func (c *LogsCmd) Run() (err error) {
 	s := NewSpinner("Getting logs from app")
+	logsHeader := "\nInstanceName[Log Type]\t\t\tLog Message\n"
+	s.FinalMSG = "done" + logsHeader
 	s.Start()
 
 	if c.Length > maxNumberLogs {
 		return fmt.Errorf("number of logs queried cannot be over %d", maxNumberLogs)
 	}
 
-	appLogs, err := api.ApplicationLogs(c.AccountID, c.AppID, c.AppPath, c.InstanceName, c.Length)
-	s.Stop()
-	if err != nil {
-		return err
+	var startTimestampRfc3339 string
+	if c.Follow {
+		log.Println("[DEBUG] Following logs...")
+		if c.InstanceName == "" {
+			return fmt.Errorf("--instance-name is required when using --follow")
+		}
+		startTimestampRfc3339 = time.Now().Format(time.RFC3339)
 	}
 
 	// Fix colorization issues between aurora and Windows
 	// https://github.com/logrusorgru/aurora#windows
 	log.SetOutput(colorable.NewColorableStdout())
 	log.SetFlags(log.Flags() &^ (log.Ldate | log.Ltime)) // Remove local time prefix on output
-	log.Printf("App InstanceName[Log Type]\t\tLog Message\n") 
-	for _, a := range appLogs {
-		a.Message = strings.TrimSpace(a.Message)
-		
-		if a.Type == "app" {
-			log.Printf("%s%s\t%s\n", aurora.Cyan(a.InstanceName), aurora.Cyan("[" + a.Type + "]"), a.Message)
-		} else if a.Type == "access" {
-			log.Printf("%s%s\t%s\n", aurora.Green(a.InstanceName), aurora.Green("[" + a.Type + "]"), a.Message)
-		} else {
-			log.Printf("%s[%s]\t%s\n", a.InstanceName, a.Type, a.Message)
+
+	for {
+		appLogs, err := api.ApplicationLogs(c.AccountID, c.AppID, c.AppPath, c.InstanceName, c.Length, startTimestampRfc3339)
+		s.Stop()
+		if err != nil {
+			return err
+		}
+		for _, a := range appLogs {
+			a.Message = strings.TrimSpace(a.Message)
+
+			if a.Type == "app" {
+				log.Printf("%s%s\t%s\n", aurora.Cyan(a.InstanceName), aurora.Cyan("["+a.Type+"]"), a.Message)
+			} else if a.Type == "access" {
+				log.Printf("%s%s\t%s\n", aurora.Green(a.InstanceName), aurora.Green("["+a.Type+"]"), a.Message)
+			} else {
+				log.Printf("%s[%s]\t%s\n", a.InstanceName, a.Type, a.Message)
+			}
+		}
+		if !c.Follow {
+			break
+		}
+		latestTimestamp := getLatestTimeStampFromLogs(appLogs)

```

Comment:
Instead of calling `getLatestTimeStampFromLogs()` and iterating the array twice, you could just track the timestamp as you emit the logs:
```suggestion
			if a.Timestamp != "" {
				latestTimestamp = a.Timestamp
			}
		}
		if !c.Follow {
			break
		}
```

Reply:
Good call. I will fix this and remove the code not required.